src/remote-config/remote-config-api-client.ts

```diff
@@ -291,6 +295,28 @@ export class RemoteConfigApiClient {
         'Remote Config conditions must be an array');
     }
   }
+
+  /**
+   * Checks if a given version number is valid.
+   * A version number must be an integer or a string in int64 format.
+   *
+   * @param {string|number} versionNumber A version number to be validated.
+   */
+  private validateVersionNumber(versionNumber: string | number): void {
+    if (!validator.isNonEmptyString(versionNumber) &&
+      !validator.isNumber(versionNumber)) {
+      throw new FirebaseRemoteConfigError(
+        'invalid-argument',
+        'versionNumber must be a non-empty string in int64 format or a number');
+    }
+    const version = validator.isNumber(versionNumber) ?
+      versionNumber : parseFloat(versionNumber as string);
```

Comment:
It looks like `Number(str)` provides stricter parsing.

Reply:
`Number` is `64 bit double` `(2^53 - 1)` so I was concerned at first as we cannot represent `int 64` `(2^63 - 1)`. But it looks like `Number.isInteger(Number(version))` should work here.